File: services/auth/src/main/java/com/example/iter/auth/dto/response/UserResponse.java
package com.example.iter.auth.dto.response;

import com.example.iter.auth.api.PreferredLanguage;
import com.example.iter.common.security.Role;
import com.example.iter.auth.domain.entity.User;
import com.example.iter.common.security.UserStatus;

public record UserResponse(
        Long id,
        String email,
        String name,
        String nickname,
        String phone,
        Role role,
        UserStatus status,
        PreferredLanguage preferredLanguage
) {
    public static UserResponse from(User user) {
        return new UserResponse(
                user.getId(),
                user.getEmail(),
                user.getName(),
                user.getNickname(),
                user.getPhone(),
                user.getRole(),
                user.getStatus(),
                user.getPreferredLanguage()
        );
    }
}

File: services/auth/src/main/kotlin/com/example/iter/auth/dto/response/UserResponse.kt
package com.example.iter.auth.dto.response

import com.example.iter.auth.api.PreferredLanguage
import com.example.iter.auth.domain.entity.User
import com.example.iter.common.security.Role
import com.example.iter.common.security.UserStatus

@JvmRecord
data class UserResponse(
    val id: Long?,
    val email: String,
    val name: String,
    val nickname: String?,
    val phone: String?,
    val role: Role,
    val status: UserStatus,
    val preferredLanguage: PreferredLanguage,
) {
    companion object {
        @JvmStatic
        fun from(user: User): UserResponse =
            UserResponse(
                user.id,
                user.email,
                user.name,
                user.nickname,
                user.phone,
                user.role,
                user.status,
                user.preferredLanguage,
            )
    }
}
